homeassistant/components/emonitor/__init__.py

```diff
@@ -25,7 +25,7 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     session = aiohttp_client.async_get_clientsession(hass)
     emonitor = Emonitor(entry.data[CONF_HOST], session)
 
-    coordinator = DataUpdateCoordinator(
+    coordinator: DataUpdateCoordinator = DataUpdateCoordinator(
```

Comment:
This type annotation shouldn't be needed. Something else is probably not correct if this solved a mypy error.

Reply:
I've thought the same thing, I've checked out this branch, I didn't find a way around it. I might be missing something as well in that case. If you have a suggestion, I'm keen on knowing why this happened.

Reply:
I think it may be due to that we have defined a generic class for the DataUpdateCoordinator. So we should type the `data` attribute by passing a type parameter to the `DataUpdateCoordinator` type annotation.
```py
coordinator: DataUpdateCoordinator[EmonitorStatus] = DataUpdateCoordinator(
```